prompt_toolkit/widgets/base.py

```diff
@@ -636,14 +636,6 @@ def _(event: E) -> None:
         def _(event: E) -> None:
             self._handle_enter()
 
-        @kb.add(Keys.Any)
-        def _(event: E) -> None:
-            # We first check values after the selected value, then all values.
-            for value in self.values[self._selected_index + 1:] + self.values:
-                if value[1].startswith(event.data):
```

Comment:
Instead of removing this key binding, can you replace the above ``startswith`` with:
```python
from prompt_toolkit.formatted_text import fragment_list_to_text, to_formatted_text

if fragment_list_to_text(to_formatted_text(value[1])).startswith(event.data):
```

I think that should fix the issue for now. (Maybe we could use a ``to_str`` helper in the ``formatted_text`` module.

Reply:
Good call.